### PR TITLE
Feature/audit endpoint

### PR DESCRIPTION
**Pull request by loneil**
# Description
To support SHOWCASE-654

Add an /audit endpoint to fetch lifecycle history by acronym.
The data returned (and likely the entire "lifecycle" storage model) will be adjusted later depending on history UX we are working on.

## Types of changes

 New feature (non-breaking change which adds functionality)
Documentation (non-breaking change with enhancements to documentation) 

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have checked that unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: backend/src/components/audit.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+// const {
+//   auditService
+// } = require('../services');
+const utils = require('./utils');
+
+const audit = {
+  /**
+   * Fetch a specific acronym's history details from GETOK database.
+   * @param {string} applicationAcronym - The app specifier.
+   */
```
Comment (jujaga) [inline backend/src/components/audit.js]:
```
  /**
   *  @function getHistoryByAcronym
   *  Fetch a specific acronym's history details from GETOK database.
   *  @param {string} applicationAcronym - The app specifier.
   *  @returns {object[]} An array containing the history of `applicationAcronym`.
   */
```

File: backend/src/components/audit.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  getHistoryByAcronym: async (applicationAcronym) => {
```
Comment (jujaga) [inline backend/src/components/audit.js]: Single parameter function doesn't need parentheses

File: backend/src/routes/v1/audit.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
+const log = require('npmlog');
+
+const audit = require('express').Router();
+const permissionHelpers = require('../../components/permissionHelpers');
+const auditComponent = require('../../components/audit');
+
+// fetches audit history for an application
+audit.get('/:appAcronym', [
+], async (req, res) => {
```
Comment (jujaga) [inline backend/src/routes/v1/audit.js]: Likely don't need the empty array as one of the middleware callbacks.

**Comment by bcgov-nr-csst** [comment]
SonarQube Analysis: https://sonarqube-k8vopl-tools.pathfinder.gov.bc.ca/dashboard?id=nr-get-token-pr-114

**Pull request merged**